### PR TITLE
Item 9012: Update application navigation menu behavior

### PR DESCRIPTION
**Pull request by labkey-susanh**
#### Rationale
We are simplifying the product/application navigation menu to better suit the majority use case for this menu.  In particular, we are removing the option to navigate to products in a different container from this menu, with the only exception being navigating to the Home folder.  

#### Related Pull Requests
* https://github.com/LabKey/platform/pull/2343
* https://github.com/LabKey/biologics/pull/909
* https://github.com/LabKey/sampleManagement/pull/594
* https://github.com/LabKey/labkey-ui-components/pull/555

#### Changes
* Remove project navigation component
* Update locators
* Add locator for new link in application menu footer
* Update LookAndFeelSettings page for new setting related to application menu.

### CODE REVIEW

File: src/org/labkey/test/components/ui/navigation/apps/LeafNavContainer.java [hunk 1]
```diff
@@ -33,14 +34,30 @@ public void clickItem(String itemText)
...
+            while(currentUrl.equals(getDriver().getCurrentUrl()) && (tries < 10))
+            {
+                wait(wait/10);
+                tries++;
+            }
+        }
+        catch (InterruptedException e)
+        {
+            throw new NotFoundException("Navigation to " + itemText + " was interrupted.");
+        }
```
Comment (labkey-danield) [inline src/org/labkey/test/components/ui/navigation/apps/LeafNavContainer.java]: Would this be cleaner?
```
        String currentUrl = getDriver().getCurrentUrl();
        WebDriverWrapper.waitFor(()->!currentUrl.equals(getDriver().getCurrentUrl()), "Did not navigate, or some such thing.", 1_000);

```

File: src/org/labkey/test/components/ui/navigation/apps/ProductListItem.java [hunk 2]
```diff
@@ -44,6 +44,11 @@ public void clickNavIcon()
         elementCache().navIcon.click();
     }
 
+    public boolean isEnabled()
+    {
+        return !"labkey-page-nav-disabled".equals(_el.getAttribute("class"));
```
Comment (labkey-danield) [inline src/org/labkey/test/components/ui/navigation/apps/ProductListItem.java]: Do we have to worry about the class attribute having anything more than "labkey-page-nav-disabled"? If so might change this to:
`return !_el.getAttribute("class").contains("labkey-page-nav-disabled");`

**Review by labkey-danield — APPROVED** [review]
I only had a couple of comments, @labkey-chrisj might have more input since he is the developer of the nav library code.

File: src/org/labkey/test/components/ui/navigation/apps/LeafNavContainer.java (hunk 1 above)
Comment (labkey-susanh) [reply src/org/labkey/test/components/ui/navigation/apps/LeafNavContainer.java]: Yes, much better.  Thanks.

File: src/org/labkey/test/components/ui/navigation/apps/ProductListItem.java (hunk 2 above)
Comment (labkey-susanh) [reply src/org/labkey/test/components/ui/navigation/apps/ProductListItem.java]: Currently, equals works, but checking contains is probably better for future-proofing.

**Pull request merged**